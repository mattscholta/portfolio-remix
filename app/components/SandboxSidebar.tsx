import classnames from "classnames";

export interface SandboxSidebarProps {
  className?: string;
}

/**
 * @name SandboxSidebar
 * @description TODO: Add a description
 */
export const SandboxSidebar = (props: SandboxSidebarProps) => {
  const { className } = props;

  // Hooks

  // Setup

  // Handlers

  // Markup

  // Life Cycle

  // ðŸ”Œ Short Circuits

  return (
    <div className={classnames("p-4", className)}>
      <h2>ðŸ”— Sandbox Sidebar</h2>
    </div>
  );
};
