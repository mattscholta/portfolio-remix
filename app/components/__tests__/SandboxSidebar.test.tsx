import { render } from '@testing-library/react';
import type { RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { describe, expect, test } from 'vitest';
import { SandboxSidebar } from '../SandboxSidebar';
import type { SandboxSidebarProps } from '../SandboxSidebar';

describe('SandboxSidebar Component', () => {
  let component: RenderResult;
  let props: SandboxSidebarProps;

  beforeEach(() => {
    props = {};

    component = render(
      <MemoryRouter>
        <SandboxSidebar {...props} />
      </MemoryRouter>,
    );
  });

  test('should render', () => {
    expect(component.baseElement).toMatchSnapshot();
  });
});
