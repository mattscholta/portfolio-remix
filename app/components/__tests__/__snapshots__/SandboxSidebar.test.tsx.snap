// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SandboxSidebar Component > should render 1`] = `
<body>
  <div>
    <div
      class="p-4"
    >
      <h2>
        ğŸ”— Sandbox Sidebar
      </h2>
    </div>
  </div>
</body>
`;
